tests/core/kernel-base-gadget-pair-single-reboot/task.yaml

```diff
@@ -18,56 +17,85 @@ environment:
     SNAP_ONE/kernel_gadget: pc
     SNAP_TWO/kernel_gadget: pc-kernel
 
-    BLOB_DIR: $(pwd)/fake-store-blobdir
+    GADGET_CHANNEL: edge
+    CORE18_GADGET_REPACK: true
+
+    # We are uploading too much in terms of size for the snaps
+    # we sideload. On UC18 this is disabled automatically, but
+    # on UC20+ this has proved to be an issue
+    SNAPD_NO_MEMORY_LIMIT: 1
 
 prepare: |
-    if [ "$TRUST_TEST_KEYS" = "false" ]; then
-        echo "This test needs test keys to be trusted"
-        exit
-    fi
+    # To get VERSION_ID defined
+    . /etc/os-release
 
-    core_snap=core20
-    if os.query is-core18; then
-        core_snap=core18
-    fi
+    # does not work for 16, but we don't support 16 for this
+    core_snap="core${VERSION_ID}"
 
     # read all for simplification
     readlink /snap/pc-kernel/current > pc-kernel.rev
     readlink "/snap/$core_snap/current" > core.rev
-    readlink "/snap/pc/current" > pc.rev
+    readlink /snap/pc/current > pc.rev
 
-restore: |
-    if [ "$TRUST_TEST_KEYS" = "false" ]; then
-        echo "This test needs test keys to be trusted"
-        exit
-    fi
-    "$TESTSTOOLS"/store-state teardown-fake-store "$BLOB_DIR"
+debug: |
+    change_id="$(cat refresh-change-id)"
+    snap debug state --change "$change_id" /var/lib/snapd/state.json
 
 execute: |
-    if [ "$TRUST_TEST_KEYS" = "false" ]; then
-        echo "This test needs test keys to be trusted"
-        exit
-    fi
+    # To get VERSION_ID defined
+    . /etc/os-release
 
-    core_snap=core20
-    if os.query is-core18; then
-        core_snap=core18
+    if not snap list pc; then
+        echo "This test needs a host using 'pc' gadget snap"
+        exit 1
     fi
 
+    # does not work for 16, but we don't support 16 for this
+    core_snap="core${VERSION_ID}"
+    
+    # correct the naming based on the system we are testing
     if [ "$SNAP_ONE" = "core" ]; then
         SNAP_ONE=$core_snap
     elif [ "$SNAP_TWO" = "core" ]; then
         SNAP_TWO=$core_snap
     fi
 
     if [ "$SPREAD_REBOOT" = 0 ]; then
-        snap ack "$TESTSLIB/assertions/testrootorg-store.account-key"
-        "$TESTSTOOLS"/store-state setup-fake-store "$BLOB_DIR"
+        echo Downloading snaps needed
+        snap download pc-kernel --channel="$VERSION_ID"/edge
 
-        "$TESTSTOOLS"/store-state init-fake-refreshes "$BLOB_DIR" "$SNAP_ONE"
-        "$TESTSTOOLS"/store-state init-fake-refreshes "$BLOB_DIR" "$SNAP_TWO"
+        snap download "$core_snap" --edge
+        unsquashfs -d base-snap "${core_snap}"_*.snap
+        rm "${core_snap}"_*
+        snap pack base-snap
```

Comment:
the only purpose of repacking is to make the snap unasserted? if so would eg `prinft '\0' >> core-for-repack.snap` be enough? it would change the checksum, but I think it should still be possible to use the snap file, at least my local experiments show that `unsquashfs` is happy extracting it and I can still mount it